### PR TITLE
SYCL variants for a couple more Basic kernels

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR adds SYCL variants for Basic kernels: ARRAY_OF_PTRS and COPY8.

This PR resolves https://github.com/LLNL/RAJAPerf/issues/431

### CODE REVIEW

**Review by homerdin — APPROVED** [review]
It looks good to me.  Tested on Intel GPU and everything is working.

File: src/basic/ARRAY_OF_PTRS-Sycl.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  const Index_type iend = getActualProblemSize();
+
+  auto res{getSyclResource()};
+  auto qu = res.get_queue();
+
+  ARRAY_OF_PTRS_DATA_SETUP;
+
+  if ( vid == Base_SYCL ) {
+
+    ARRAY_OF_PTRS_Array x_array = x;
```
Comment (MrBurmark) [inline src/basic/ARRAY_OF_PTRS-Sycl.cpp:39]: Why are we doing this here?

Comment (rhornung67) [reply src/basic/ARRAY_OF_PTRS-Sycl.cpp:39]: I don't understand the question. Isn't this following the same implementation pattern as the CUDA version?

Comment (artv3) [reply src/basic/ARRAY_OF_PTRS-Sycl.cpp:39]: I think question is why doesn't the RAJA variant require it?

Comment (MrBurmark) [reply src/basic/ARRAY_OF_PTRS-Sycl.cpp:39]: I remember now, when you try to pass an array as a function argument it just passes a pointer, so for cuda and hip where we have a global function we wrap the array in a struct to pass it by value. Here where it is captured in a lambda that is not necessary.

File: src/basic/ARRAY_OF_PTRS-Sycl.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+      const size_t global_size = work_group_size * RAJA_DIVIDE_CEILING_INT(iend, work_group_size);
+
+      qu->submit([&] (sycl::handler& h) {
+        h.parallel_for(sycl::nd_range<1>(global_size, work_group_size),
+                       [=] (sycl::nd_item<1> item ) {
+
+          Index_type i = item.get_global_id(0);
+          if (i < iend) {
+            ARRAY_OF_PTRS_BODY(x_array.array);
```
Comment (rhornung67) [inline src/basic/ARRAY_OF_PTRS-Sycl.cpp:52]:
```suggestion
            ARRAY_OF_PTRS_BODY(x);
```

@MrBurmark so remove the line above and do this?

Comment (MrBurmark) [reply src/basic/ARRAY_OF_PTRS-Sycl.cpp:52]: Ya, that should do it.

**Comment by rhornung67** [comment]
> It looks good to me. Tested on Intel GPU and everything is working.

Thanks @homerdin I'm testing on an AMD-based system with a special Intel compiler build we made.

**Review by MrBurmark — APPROVED** [review]

**Pull request merged**